packages/zql/src/ivm/fan-in.ts

```diff
@@ -58,8 +60,15 @@ export class FanIn implements Operator {
     return this.#schema;
   }
 
-  fetch(req: FetchRequest): Stream<Node> {
-    return this.#fetchOrCleanup(input => input.fetch(req));
+  *fetch(req: FetchRequest): Stream<Node> {
+    assert(this.#accumulatedPushes.length === 0);
+    for (const node of this.#fanOut.fetch(req)) {
+      // Did any of the branches push the synthetic add change through?
+      if (this.#accumulatedPushes.length) {
+        yield node;
+      }
+      this.#accumulatedPushes = [];
```

Comment:
Why is this outside the if? Can we accumulate more pushes while yielding the node above?

If we can that seems very problematic for this entire approach as it means multiple fetches can be passing through this node concurrently.